src/dialogs/addCategory.js

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from "react";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import FormControl from "@mui/material/FormControl";
+import { Form, Formik } from "formik";
+import * as Yup from "yup";
+import { DialogTitle, Dialog, IconButton } from "@mui/material";
+import CloseIcon from "../icons/closeIcon";
+
+const styles = {
+  title: {
+    color: "#1565C0",
+    fontWeight: "bold",
+  },
+  displayCenter: {
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  addMore: {
+    backgroundColor: "#2CB23B",
+    color: "white",
+    mb: 2,
+  },
+  submit: {
+    letterSpacing: "2px",
+    fontWeight: "bold",
+    mb: 2,
+  },
+};
+
+export default function AddCategory(props) {
+  const { open, handleClose, addItem } = props;
+  const [products, setProducts] = useState(["product1"]);
+
+  const validationSchema = Yup.object({
+    categoryName: Yup.string().required("Category Name is required"),
+  });
+
+  const onSubmit = async (values) => {
+    if (values.categoryName) {
+    let newProducts = []; 
+    products.forEach((product, index) => {
+      if (values[product]) {
+        const data = {
+          id: index + 1,
+          name: values[product],
+        };
+        newProducts.push(data);
+      }
+    });
+    addItem(values.categoryName.trim().length? values.categoryName : "No Name", newProducts);
+    handleClose();
+    }
+  };
+
+  const addMoreProduct = () => {
+    setProducts([...products, `product${products.length + 1}`]);
+  };
+
+  return (
+    <Dialog open={open} fullWidth={true} maxWidth="sm">
+      <DialogTitle sx={styles.title}>
+        {"Add New Category"}
```

Comment:
Try not to use JSX tags for simple strings, This can be re-written as Add New Category